storage/types.go

```diff
@@ -412,9 +405,13 @@ func (n *NodeID) MaskLeft(depth int) *NodeID {
 // Neighbor returns the same node with the bit at PrefixLenBits flipped.
 // In terms of a tree traversal, this is the parent node's other child node
 // in the binary tree (often termed sibling node).
+// TODO(Martin2112): This is only used by Siblings() in conjunction with
+// MaskLeft() and could be cleaned up further.
 func (n *NodeID) Neighbor() *NodeID {
 	height := n.PathLenBits() - n.PrefixLenBits
-	return n.FlipRightBit(height)
+	bIndex := (n.PathLenBits() - height - 1) / 8
```

Comment:
Maybe add a comment that this is an inlined FlipRightBit function?

Reply:
For context, I found all of these manipulation functions very hard to understand and so I broke them down into the smallest functions I could grok and unit test.

Reply:
These are internal APIs that most users won't see. That said the documentation wasn't as clear as it could be.

The goal is to remove all functions that mutate node ids in place, which prevents their representations from being shared and blocks storage optimizations.